src/iceberg/util/formatter.h

```diff
@@ -40,3 +43,102 @@ struct std::formatter<Derived> : std::formatter<std::string_view> {
     return std::formatter<string_view>::format(obj.ToString(), ctx);
   }
 };
+
+/// \brief std::formatter specialization for std::vector
+template <typename T>
+struct std::formatter<std::vector<T>> : std::formatter<std::string_view> {
```

Comment:
FWIW, I _think_ the "idiomatic" way is supposed to be `std::format::join` and ranges. I wonder if specializing on std types may come back to bite us later.

Example: https://stackoverflow.com/a/77993463

Reply:
Or another example, albeit C++23: https://stackoverflow.com/a/71196142

Reply:
I think we need to rename this file to `formatter_internal.h` so we don't expose it to the downstream. In this approach, we take full control of it.

Reply:
SGTM.

It might be good to expose formatter specializations on our own types, just so long as they're separate from the specializations for std types

Reply:
Ah ok I see that's what you've done.